backend/editor/entries.py

```diff
@@ -78,34 +82,63 @@ def import_from_github(self, description):
         filename = self.taxonomy_name + '.txt'
         base_url += filename
         try:
-            # Downloads and creates taxonomy file in current working directory
-            urllib.request.urlretrieve(base_url, filename)
-            status = self.parse_taxonomy(filename) # Parse the taxonomy
-            self.create_project(description) # Creates a "project node" in neo4j
-            os.remove(filename) # Removes taxonomy file after parsing
-            return status
+            with tempfile.TemporaryDirectory(prefix="taxonomy-") as tmpdir:
+
+                # File to save the downloaded taxonomy
+                filepath = f"{tmpdir}/{filename}"
+
+                # Downloads and creates taxonomy file in current working directory
+                urllib.request.urlretrieve(base_url, filepath)
+
+                status = self.parse_taxonomy(filepath) # Parse the taxonomy
+                self.create_project(description) # Creates a "project node" in neo4j
+                return status
         except:
-            return TaxnonomyImportError()
+            raise TaxnonomyImportError()
     
-    def export_taxonomy(self):
+    def export_taxonomy(self, type_of_export):
         """
-        Helper function to export a taxonomy for download
+        Helper function to export a taxonomy through download or Github
         """
         # Close current transaction to use the session variable in unparser
         get_current_transaction().commit()
 
+        self.close_project() # Close the project
+
         # Create unparser object and pass current session to it
         unparser_object = unparser.WriteTaxonomy(get_current_session())
         filename = self.taxonomy_name + '.txt'
         try:
             # Parse taxonomy with given file name and branch name
             unparser_object(filename, self.branch_name, self.taxonomy_name)
-            return filename
+            if type_of_export == "download":
+                return filename
+            elif type_of_export == "github":
+                return self.export_to_github(filename)
         except:
-            return TaxonomyUnparsingError()
-
+            raise TaxonomyUnparsingError()
     
-    def check_if_project_exists(self):
+    def export_to_github(self, filename):
+        """
+        Helper function to export a taxonomy to GitHub
+        """
+        query = """MATCH (n:PROJECT) WHERE n.id = $project_name RETURN n.description"""
+        result = get_current_session().run(query, {"project_name": self.project_name})
+        description = result.data()[0]['n.description']

```

Comment:
Note: in the future, we could add a link to the taxonomy editor project so that reviewer can check it there :-)